### PR TITLE
Fix rails 5 payments

### LINKED ISSUE #7004: StripeSCA: Error 400 / Invalid paymeny state when checking out with a 3D-secure card
## Description

Checking out with Stripe SCA, using a 3D secure card which always requires authentication like `4000 0027 6000 3184` leads to error 400, on the checkout screen.

Checking out with a regular card (424242...) works-

## Expected Behavior

The authentication step should follow (in staging)

## Actual Behaviour

Error 400 / Bad request.

## Steps to Reproduce

1. Admin: Set up stripe SCA payment method in the backoffice (this worked fine)
2. Customer: add some items to the cart and attempt checking out with the SCA
3. Insert the card `4000 0027 6000 3184` -> See the error Invalid Payment state
4. Now try checking out with the 424242... card.

## Animated Gif/Screenshot

image

## Your Environment

* Version used: rails 5.0 branch
* Browser name and version: Firefox 86
* Operating System and version (desktop or mobile): Desktop Ubuntu 20.04

### LINKED ISSUE #7047: Backoffice/StripeSCA: Error when creating payments with a 3D-secure card
## Description

Creating payments in the backoffice is failing for Stripe-SCA cards, like `x-3184`.

Maybe related to #7004 .

## Expected Behavior

An email requiring authentication should be sent to the customer; payment state should be 'pending' till authentication completed.

## Actual Behaviour

Payment fails.

## Steps to Reproduce

1. Create an order in the BO.
2. Create a new payment, for that order
3. Select stripe SCA payment method and insert the required data for card `x-3184`.

### PR DESCRIPTION
**Pull request by andrewpbrett**
#### What? Why?

Closes #7004 
Closes #7047 

#### Release notes

Changelog Category: User facing changes | Technical changes

### CODE REVIEW

File: app/controllers/spree/admin/payments_controller.rb [hunk 1]
```diff
@@ -153,7 +155,7 @@ def load_payment
       def authorize_stripe_sca_payment
         return unless @payment.payment_method.class == Spree::Gateway::StripeSCA
 
-        @payment.authorize!(main_app.order_path(@payment.order, only_path: false))
```
Comment (sauloperez) [inline app/controllers/spree/admin/payments_controller.rb:156]: isn't this what `main_app.order_url` does?

**Comment by filipefurtad0** [comment]
Hey @andrewpbrett thanks for this PR and @sauloperez for reviewing.

The "happy-path" order placement and payment scenarios with a 3D-card (x-3184) were tested now:

- Shopfront - OK
- Backoffice - OK

Since we are going for rails 5 with this, I think this may need more extensive testing. Should testing wait for further reviews/commits?

**Comment by sauloperez** [comment]
> Should testing wait for further reviews/commits?

not for me. Mine is a minor comment that could be solved separately if we think is worth it. Gogogogo!

**Comment by filipefurtad0** [comment]
Allright!

Tested the triggering common errors, for example, failing a payment or using a card with insufficient funds (x-3178):
- Frontoffice, all good here
- Backoffice, met issue #6964 (s5, not rails 5 related)

Tested subscriptions, with SCA payments (with cards which require auth)
- all good

Ready to go!

**Pull request merged**